### PR TITLE
Improve `Stitch1D` documentation

### LINKED ISSUE #18117: Improve documentation for `Stitch1D`
This issue is to improve documentation for `Stitch1D` to explain how the overlap region, scale factor, etc are calculated. We should include an explanation on how errors are calculated.

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
Adds some improvements to `Stitch1D` documentation.

**To test:**

Build the documentation and make sure it describes accurately what the algorithm is doing. Specially, make sure that the note about how errors are propagated is correct.

Fixes #18117.

Release notes have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/Stitch1D-v3.rst [hunk 1]
```diff
@@ -10,27 +10,62 @@ Description
...
+#. The weighted mean of the two workspaces in range [:literal:`StartOverlap`, :literal:`EndOverlap`]
+   is calculated. Note that if both workspaces have zero errors, an un-weighted mean will be
+   performed instead.
+#. The output workspace will be created by summing the left-hand-side workspace (values in range
+   [:literal:`StartX`, :literal:`StartOverlap`], where :literal:`StartX` is the minimum X value
+   specified via :literal:`Params` or calculated from the left-hand-side workspace) + weighted
+   mean workspace + right-hand-side workspace (values in range [:literal:`EndOverlap`, :literal:`EndX`],
+   where :literal:`EndX` is the maximum X value specified via :literal:`Params` or calculated
+   from the right-hand-side workspace) multiplied by the scale factor.
+#. The special values are put back in the output workspace.
```
Comment (gvardany) [inline docs/source/algorithms/Stitch1D-v3.rst]: How are they put in the overlap region?

**Review by gvardany — COMMENTED** [review]
It reads quite clear. I had one question (see inline), not sure if this is important though.

File: docs/source/algorithms/Stitch1D-v3.rst (hunk 1 above)
Comment (raquelalvarezbanos) [inline docs/source/algorithms/Stitch1D-v3.rst]: In step 2, we record the indexes that correspond to special values. Essentially what we have is four vectors, `m_nanYIndexes`, `m_infYIndexes`, `m_nanEIndexes` and `m_infEIndexes`, that record the bin indexes in which NaN and infinite values were found (both for errors and Y values). Then we replace those bins in the output workspace with the corresponding special value, i.e. NaN or infinite.

Comment (raquelalvarezbanos) [inline docs/source/algorithms/Stitch1D-v3.rst]: I think scientists are more worried about how errors are propagated, as they were not sure that they are propagated correctly. For instance, they didn't know if the scale factor (when calculated by the algorithm) had an error associated. From the code, I think it does have an error associated and this error is propagated correctly.

Comment (gvardany) [inline docs/source/algorithms/Stitch1D-v3.rst]: Yes, that's clear, but I was wondering how this is done in the overlap region. Imagine in one overlap bin there is NaN in LHS, but Inf in RHS, what is put back?

Comment (raquelalvarezbanos) [inline docs/source/algorithms/Stitch1D-v3.rst]: Sorry, I didn't understand you question. That's a good point, I didn't think about that case. From the code I see that in that case we'll put an `Inf` in the output workspace. Not sure if this is the best way to deal with this case, but it is done like that. I think I am going to add a sentence about this so it's clear in the documentation. Thanks!

**Comment by gvardany** [comment]
Ok, it's better now. When tests pass :shipit:

**Pull request merged**